service/matching/ack_manager.go

```diff
@@ -112,29 +128,41 @@ func (m *ackManager) setAckLevel(ackLevel int64) {
 	}
 }
 
-func (m *ackManager) completeTask(taskID int64) (ackLevel int64) {
+func (m *ackManager) completeTask(taskID int64) int64 {
 	m.Lock()
 	defer m.Unlock()
-	if completed, ok := m.outstandingTasks[taskID]; ok && !completed {
-		m.outstandingTasks[taskID] = true
-		m.backlogCounter.Dec()
+	macked, found := m.outstandingTasks.Get(taskID)
+	acked := macked.(bool)
+	if !found || acked {
+		// don't adjust ack level if nothing has changed
+		return m.ackLevel
 	}
 
-	// TODO the ack level management should be done by a dedicated coroutine
-	//  this is only a temporarily solution
```

Comment:
I think we need to keep this TODO. 

The ack level management and task deletion can be done in the background periodically or triggered when there are too many tasks pending and blocking read. I think in existing implementation the returned ack level is also ignored by the taskGC component most of the time.